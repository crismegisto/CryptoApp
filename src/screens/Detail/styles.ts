import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  loading: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  container: {
    backgroundColor: 'white',
    margin: 6,
    padding: 10,
    borderRadius: 10
  },
  subContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  icon: {
    width: 100,
    height: 100,
    resizeMode: 'center'
  }
});

export default styles;
