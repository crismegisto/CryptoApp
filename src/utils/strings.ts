export const isStringMatch = (query: string | undefined, target: string): boolean => {
  const queryLowerCase = query?.toLowerCase();
  return queryLowerCase ? target.toLowerCase().includes(queryLowerCase) : false;
};
