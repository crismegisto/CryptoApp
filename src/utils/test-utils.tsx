import React, { PropsWithChildren } from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import type { PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import { AppStore, RootState, setupStore } from '../store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  { preloadedState = {}, store = setupStore(preloadedState), ...renderOptions }: ExtendedRenderOptions = {}
) {
  const wrapper = ({ children }: PropsWithChildren<{}>): JSX.Element => (
    <Provider store={store}>{children}</Provider>
  );

  return { store, ...render(ui, { wrapper, ...renderOptions }) };
}

export * from '@testing-library/react-native';

export { renderWithProviders as render };
