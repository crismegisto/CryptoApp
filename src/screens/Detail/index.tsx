import { View, Text, Image } from 'react-native';
import React from 'react';
import { RouteProp, useRoute } from '@react-navigation/native';

import { RoutesParamList } from '../../constants/routesParamList';
import { useFetchSpecifCoinQuery } from '../../services/apiSlice';
import { Coin } from '../../interfaces';
import { icons } from '../../constants/icons';
import RowDetail from '../../components/RowDetail';
import styles from './styles';

const Detail = () => {
  const { params } = useRoute<RouteProp<RoutesParamList>>();
  const { data = {} as Coin, isLoading } = useFetchSpecifCoinQuery(params?.id || '90');

  const {
    rank,
    name,
    symbol,
    price_usd: price,
    percent_change_1h: $1h,
    percent_change_7d: $7h,
    percent_change_24h: $24h,
    market_cap_usd: marketCap,
    volume24: volume,
    csupply
  } = data;

  if (isLoading) {
    return (
      <View style={styles.loading}>
        <Text>Loading...</Text>
      </View>
    );
  }
  return (
    <View style={styles.container}>
      <View style={styles.subContainer}>
        <View>
          <RowDetail label="#" description={rank} />
          <RowDetail label="Name" description={name} />
          <RowDetail label="Symbol" description={symbol} />
          <RowDetail label="Price" description={price} dollarSign />
          <RowDetail label="1h%" description={$1h} percentSign />
          <RowDetail label="7h%" description={$7h} percentSign />
          <RowDetail label="24h%" description={$24h} percentSign />
          <RowDetail label="Volume" description={volume} dollarSign />
          <RowDetail label="Market Cap" description={marketCap} dollarSign />
          <RowDetail label="Circulation Supply" description={csupply} dollarSign />
        </View>
        <Image style={styles.icon} source={icons[symbol] ? icons[symbol] : icons.BTC} />
      </View>
    </View>
  );
};

export default Detail;
