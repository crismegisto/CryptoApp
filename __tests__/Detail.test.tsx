import React from 'react';

import Detail from '../src/screens/Detail';
import { render, screen } from '../src/utils/test-utils';

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useRoute: () => ({ params: { id: '90' } })
  };
});

jest.useFakeTimers();
test('snapshot', async () => {
  render(<Detail />);
  expect(screen.toJSON()).toMatchSnapshot();
  expect(await screen.findByText('Market Cap')).toBeTruthy();
});
