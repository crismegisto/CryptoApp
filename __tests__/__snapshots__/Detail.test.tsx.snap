// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Loading...
  </Text>
</View>
`;
